blog post: write blog post about the 1.23 release (#320)
kubewarden/kubewarden.io: content/blog/2025/03/kubewarden-1.23-release.md

---
title: Kubewarden 1.23 release
authors:
  - Flavio Castelli
date: 2025-03-26
---

The wait is over—Kubewarden 1.23 has arrived! Packed with exciting security enhancements,
smoother workflows, and important updates, this release is here to make your Kubernetes
experience even better. Let's dive into what’s new!

## Hardening of the admission webhooks

[Kubernetes Dynamic Admission Controllers](https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/),
like Kubewarden, work by providing a webhook server
that implements the validation/mutation API defined by the Kubernetes project.
These webhook servers are usually deployed within the same cluster as regular Kubernetes workloads.
They are exposed internally using Kubernetes' `Service` resources.

The majority of the time, the only legitimate consumer of these webhooks is the Kubernetes API
Server. However, due to [Kubernetes' network model](https://kubernetes.io/docs/concepts/services-networking/),

> any Pod running inside of the cluster can make requests against them.

Reviewer: ```suggestion
any Pod running in the cluster can make requests against them.
```

Author: I've added it manually now, sorry... I missed that earlier

This attack surface can be reduced by limiting who can actually interact with these webhook servers.

One way to reduce this attack surface is by deploying network policies, but an even stronger approach is
using mutual TLS (mTLS). With mTLS, both the client and server verify each other's identity using
certificates, ensuring secure and encrypted communication while blocking unauthorized access.

Starting from this release, all the Kubewarden webhooks can be protected using mTLS. This can be done using

> a configuration option inside of our helm charts.

Reviewer: ```suggestion
a configuration option in our helm charts.
```

Unfortunately this cannot be turned on by default, since some manual operations are required by the
administrator of the Kubernetes cluster.
[This howto](https://docs.kubewarden.io/howtos/security-hardening/webhook-mtls) explains how to enable
mTLS when deploying Kubewarden on k3s.

Finally, take a look at [this reference documentation](https://docs.kubewarden.io/reference/security-hardening/webhooks-hardening)
to learn more about how to perform security hardening of Kubewarden webhooks. This
includes information both about mTLS and reference network policies that can be used
to improve the security posture of your deployment.

## Support of PSA `restricted` profile

Kubernetes [Pod Security Admission (PSA)](https://kubernetes.io/docs/concepts/security/pod-security-admission/)
is a stable feature since the Kubernetes 1.25 release. It provides a way to validate the workloads created

> inside of a Namespace using different set of rules. These rules are hard-coded inside of Kubernetes

Reviewer: ```suggestion
in a Namespace using different set of rules. These rules are hard-coded in Kubernetes
```

> and are grouped into three different profiles, from a more relaxed to a stricter one.

Reviewer: ```suggestion
and are grouped into three different profiles, ranging from relaxed to strict.
```
